src/common/reply.h

```diff
@@ -0,0 +1,13 @@
+#ifndef REPLY_H
+#define REPLY_H
+
+// Meta definitions
+#define RPL_META_MESSAGE(servername, numeric, message) (std::string(":") + servername + " " + numeric + " " + message + "\r\n")
+
+// Numeric replies in order
+#define RPL_WELCOME_001(servername, nick, user, host) (RPL_META_MESSAGE(servername, "001", "Welcome to the Internet Relay Network " + nick + "!" + user "@" + host))
+#define RPL_YOURHOST_002(servername, version) (RPL_META_MESSAGE(servername, "002", "Your host is " + servername + ", running version " + version))
+#define RPL_CREATED_003(servername, date) (RPL_META_MESSAGE(servername, "003", "This server was created " + date))
+#define RPL_MYINFO_004(servername, version, user_modes, channel_modes) (RPL_META_MESSAGE(servername, "004", servername + " " + version + " " + user_modes + " " + channel_modes))
```

Comment:
Here the IRCv3 spec calls for the first parameter to be the client nickname, while RFC 2812 does not include the client nickname at all. 
IRCv3 also has an extra optional parameter at the end,  called ```channel modes with a parameter```.

| spec. | message parameters |
| --- | --- |
| RFC 2812 | ```<servername> <version> <available user modes> <available channel modes>``` |
| IRCv3 | ```<client> <servername> <version> <available user modes> <available channel modes> [<channel modes with a parameter>]``` |